director/handler/scan/scan.go

```diff
@@ -0,0 +1,158 @@
+// Package scan implements handler for scans
+package scan
+
+import (
+	"fmt"
+	"log"
+
+	"github.com/google/uuid"
+	"github.com/greenbone/eulabeia/connection"
+	"github.com/greenbone/eulabeia/director/handler/target"
+	"github.com/greenbone/eulabeia/messages"
+	"github.com/greenbone/eulabeia/messages/handler"
+	"github.com/greenbone/eulabeia/models"
+	"github.com/greenbone/eulabeia/storage"
+)
+
+// Storage is for poutting and getting a models.Scan
+type Storage interface {
+	Put(models.Scan) error            // Overrides existing or creates a models.Scan
+	Get(string) (*models.Scan, error) // Gets a models.Scan via ID
+	Delete(string) error
+}
+
+// Depositary stores models.Scan as json within a given StorageDir
+// The filename is a uuid without suffix.
+type Depositary struct {
```

Comment:
Probably more about style; I rather see definitions at one glance especially if they're used in the same context and don't have meaning outside of that.

Reply:
We had a discussion about it; since we are jumping between go and c and it is not that uncommon for go to have multiple structs defined we decided that we keep it for now but will further separate when:
- it's getting out of hand
- structs don't share a common domain

So although it will get redefined most likely not within this pr.